ansible_base/authentication/serializers/authenticator.py

```diff
@@ -3,25 +3,40 @@
 from django.utils.translation import gettext_lazy as _
 from rest_framework.serializers import ChoiceField, ValidationError
 
-from ansible_base.authentication.authenticator_plugins.utils import get_authenticator_plugin, get_authenticator_plugins
+from ansible_base.authentication.authenticator_plugins.utils import generate_authenticator_slug, get_authenticator_plugin, get_authenticator_plugins
 from ansible_base.authentication.models import Authenticator
 from ansible_base.lib.serializers.common import NamedCommonModelSerializer
+from ansible_base.lib.serializers.mixins import ImmutableFieldsMixin
 from ansible_base.lib.utils.encryption import ENCRYPTED_STRING
 from ansible_base.lib.utils.response import get_relative_url
 
 
-class AuthenticatorSerializer(NamedCommonModelSerializer):
+class AuthenticatorSerializer(NamedCommonModelSerializer, ImmutableFieldsMixin):
     type = ChoiceField(get_authenticator_plugins())
 
     def validate_type(self, value):
         if self.instance and self.instance.type != value:
             raise ValidationError(_("Cannot change authenticator type after it has been created."))
         return value
 
+    def validate_slug(self, value):
+        value = generate_authenticator_slug(value)
+        if self.instance and self.instance.slug != value:
+            raise ValidationError(_("Cannot change slug after it has been created."))
```

Comment:
Testing this from GW API I just see the slug not changed and the config returned as it was with a 200 and no entry in the log, so not sure if that is proper or if this validationerror should be visible.  Same for PUT and PATCH.

Reply:
That makes sense, I'll toss that validation error. It seems `read_only` prevents the field from being included in PUT/PATCH inputs.